src/utils/couponUtils.js

```
@@ -0,0 +1,193 @@
+  }
+
+  let discountAmount = 0;
+
+  switch (coupon.type) {
+    case 'percentage':
+      // 퍼센테이지 할인: 주문 금액의 %로 계산 후 100원 단위 내림
+      const rawPercentageDiscount = orderPrice * (coupon.discount / 100);
+      discountAmount = Math.floor(rawPercentageDiscount / 100) * 100;
+      
+      // 최대 할인 금액 제한 적용
+      if (coupon.maxDiscount && discountAmount > coupon.maxDiscount) {
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)